src/sentry/profiles/task.py

```diff
@@ -225,18 +225,35 @@ def _normalize(profile: Profile, organization: Organization) -> None:
 def _prepare_frames_from_profile(profile: Profile) -> Tuple[List[Any], List[Any]]:
     modules = profile["debug_meta"]["images"]
 
+    # NOTE: the usage of `adjust_instruction_addr` assumes that all
+    # the profilers on all the platforms are walking stacks right from a
+    # suspended threads cpu context
+
     # in the sample format, we have a frames key containing all the frames
     if "version" in profile:
-        stacktraces = [{"registers": {}, "frames": profile["profile"]["frames"]}]
+        frames = [dict(frame) for frame in profile["profile"]["frames"]]
+
+        for stack in profile["profile"]["stacks"]:
+            if len(stack) > 0:
+                first_frame_idx = stack[0]
+                frames[first_frame_idx]["adjust_instruction_addr"] = False
```

Comment:
Re-writing here as I submitted a comment instead of a review.

I'm not sure this is what you want. 

This adjustment must not happen for leaf frames, from what I read in the [RFC 43](https://github.com/getsentry/rfcs/blob/main/text/0043-instruction-addr-adjustment.md).

Here you're looping through the stacks and set the 0-indexed (of the stack) frame  `adjust_instruction_addr ` field to `False`.

But since we reuse the "global" `frames` list, the same frame could appear in a different stack in a non-leaf position.

Reply:
Can it actually happen that a frame is the starting frame of one stacktrace and in the middle of another stacktrace?

Reply:
The other way around, the 2nd and lower frame is always a return address, and the first frame is not. So even though that frame is shared by multiple threads / samples (for example waiting on a syscall), that frame should never be a non-leaf of a different stack.